module/VuFind/src/VuFind/RecordTab/Map.php

```diff
@@ -131,6 +138,40 @@ public function getMapGraticule()
         return $this->graticule;
     }
 
+    /**
+     * Get the basemap alias setting.
+     *
+     * @return array
+     */
+    public function getBasemap()
```

Comment:
If this code is copied and pasted in two different places, it might make sense to add something like a VuFind\Map\BasemapTrait to avoid redundancy. The two classes could then use the trait, and the only difference between them would be how they configure themselves to override the default $this->basemap.

Reply:
I'm wondering if just reading the configuration options from a .ini file would solve this issue. That way no lookup function (like getBasemap) is needed, since the settings will just be passed in from the geofeatures.ini file and if those settings are blank, we use the default settings.

Reply:
True -- we can see how the code settles out, but it's definitely possible that moving the configuration will simplify this to the point where it's trivial and the need for a trait simply goes away.